### PR TITLE
<DatePicker /> -  add clear button prop

### PR DESCRIPTION
**Pull request by bamai**
https://jira.wixpress.com/browse/DSM-1656

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-ui-tpa**

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_6890/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-style-react**

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6890

**Comment by assafzi** [comment]
UX Approved.
Thanks!

File: packages/wix-style-react/src/DatePicker/test/DatePicker.spec.js [hunk 1]
```diff
@@ -788,6 +788,67 @@ describe('DatePicker', () => {
...
+      it('should not have clearButton by default', async () => {
+        const onChange = jest.fn();
+        const {
+          driver: { inputDriver },
+        } = render(<DatePicker onChange={onChange} />);
+
+        expect(await inputDriver.hasClearButton()).toBe(false);
+      });
+
+      it('should not have clearButton when value is not selceted', async () => {
```
Comment (mykas) [inline packages/wix-style-react/src/DatePicker/test/DatePicker.spec.js:801]:
```suggestion
      it('should not have clearButton when value is not selected', async () => {
```

File: packages/wix-style-react/src/DatePicker/test/DatePicker.spec.js [hunk 2]
```diff
@@ -788,6 +788,67 @@ describe('DatePicker', () => {
...
+        const {
+          driver: { inputDriver },
+        } = render(
+          <DatePicker onChange={onChange} clearButton onClear={onClear} />,
+        );
+
+        expect(await inputDriver.hasClearButton()).toBe(false);
+      });
+
+      it('should have clearButton when value is selceted', async () => {
```
Comment (mykas) [inline packages/wix-style-react/src/DatePicker/test/DatePicker.spec.js:813]:
```suggestion
      it('should have clearButton when value is selected', async () => {
```

**Review by mykas — APPROVED** [review]
Looks amazing @bamai ! Very clean and simple to understand :)

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

A deploy preview of your PR was successfully released 🖼.

Project Name: **wix-ui-tpa**

You can find it under one of the following URLs:
* https://bo.wix.com/pages/wix-ui-tpa.pr_6890/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

A deploy preview of your PR was successfully released 🖼.

Project Name: **wix-style-react**

You can find it under one of the following URLs:
* https://preview.wix-style-react.com/pr_6890

**Pull request merged**